fido2/win_api.py

```diff
@@ -0,0 +1,730 @@
+        ("pvExtension", PBYTE),
+    ]
+
+
+class WebAuthNExtensions(ctypes.Structure):
+    """Maps to WEBAUTHN_EXTENSIONS Struct.
+
+    https://github.com/microsoft/webauthn/blob/master/webauthn.h#L324
+    """
+
+    _fields_ = [
+        ("cExtensions", DWORD),
+        ("pExtensions", ctypes.POINTER(WebAuthNExtension)),
+    ]
+
+
+class WebAuthNCredential(ctypes.Structure):
+    """Maps to WEBAUTHN_CREDENTIAL Struct.
+
+    https://github.com/microsoft/webauthn/blob/master/webauthn.h#L212
+
+    :param Dict[str, Any] cred: Dict of Credential Descriptor data.
+    """
+
+    _fields_ = [
+        ("dwVersion", DWORD),
+        ("cbId", DWORD),
+        ("pbId", PBYTE),
+        ("pwszCredentialType", LPCWSTR),
+    ]
+
+    id = BytesProperty("Id")
+
+    def __init__(self, cred):
+        self.id = cred["id"]
+        self.pwszCredentialType = cred["type"]
+
+    @property
+    def descriptor(self):
+        return {"type": self.pwszCredentialType, "id": self.id}
+
+
+class WebAuthNCredentials(ctypes.Structure):
+    """Maps to WEBAUTHN_CREDENTIALS Struct.
+
+    https://github.com/microsoft/webauthn/blob/master/webauthn.h#L219
+
+    :param List[Dict[str, Any]] credentials: List of dict of
+        Credential Descriptor data.
+    """
+
+    _fields_ = [
+        ("cCredentials", DWORD),
+        ("pCredentials", ctypes.POINTER(WebAuthNCredential)),
+    ]
+
+    def __init__(self, credentials):
+        self.cCredentials = len(credentials)
+        self.pCredentials = (WebAuthNCredential * len(credentials))(
+            *(WebAuthNCredential(cred) for cred in credentials)
+        )
+
+
+class WebAuthNGetAssertionOptions(ctypes.Structure):
+    """Maps to WEBAUTHN_AUTHENTICATOR_GET_ASSERTION_OPTIONS Struct.
+
+    https://github.com/microsoft/webauthn/blob/master/webauthn.h#L452
+
+    :param int timeout: Time that the operation is expected to complete within.
+        This is used as guidance, and can be overridden by the platform.
+    :param WebAuthNAuthenticatorAttachment attachment: Platform vs Cross-Platform
+        Authenticators.
+    :param WebAuthNUserVerificationRequirement user_verification_requirement: User
+        Verification Requirement.
+    :param List[Dict[str,Any]] credentials: Allowed Credentials List.
+    """
+
+    _fields_ = [
+        ("dwVersion", DWORD),
+        ("dwTimeoutMilliseconds", DWORD),
+        ("CredentialList", WebAuthNCredentials),
+        ("Extensions", WebAuthNExtensions),
+        ("dwAuthenticatorAttachment", DWORD),
+        ("dwUserVerificationRequirement", DWORD),
+        ("dwFlags", DWORD),
+        ("pwszU2fAppId", PCWSTR),
+        ("pbU2fAppId", BOOL),
+        ("pCancellationId", ctypes.POINTER(GUID)),
+        ("pAllowCredentialList", ctypes.POINTER(WebAuthNCredentialList)),
+    ]
+
+    def __init__(self, timeout, attachment, user_verification_requirement, credentials):
+        self.dwVersion = get_version(self.__class__.__name__)
+        self.dwTimeoutMilliseconds = timeout
```

Comment:
can this be made to convert seconds (int of float) into milliseconds for feature parity with the other clients?

```suggestion
        self.dwTimeoutMilliseconds = int(timeout * 1000)
```

for `WebAuthNMakeCredentialOptions` as well.

Reply:
This has been resolved by going the other way, the timeout is now given in ms as per the WebAuthn spec in both Client implementations.